app/components/Modal/DepositModal.jsx

```diff
@@ -67,28 +67,36 @@ class DepositModalContent extends DecimalChecker {
         return !utils.are_equal_shallow(ns, this.state);
     }
 
-    onGatewayChanged(e) {
-        if (!e.target.value) return;
-        this._getDepositAddress(this.state.selectedAsset, e.target.value);
+    onGatewayChanged(value) {
+        if (!value) return;
+        this._getDepositAddress(this.state.selectedAsset, value);
     }
 
-    onAssetSelected(asset, assetDetails) {
-        if (assetDetails.gateway == "")
-            return this.setState({selectedAsset: asset, selectedGateway: null});
-
-        let {selectedAsset, selectedGateway} = _onAssetSelected.call(
-            this,
-            asset,
-            "depositAllowed",
-            (availableGateways, balancesByGateway) => {
-                if (availableGateways && availableGateways.length == 1)
-                    return availableGateways[0]; //autoselect gateway if exactly 1 item
-                return null;
-            }
-        );
+    onAssetSelected(coinItems, selected) {
+        let asset;
```

Comment:
Is asset always to be found in coinItems? If not, please add gracefull error handling, or comment why its not necessary